app/commands/exercise/update_median_wait_times.rb

```diff
@@ -0,0 +1,29 @@
+class Exercise
+  class UpdateMedianWaitTimes
+    include Mandate
+
+    def call
+      Exercise.find_each do |exercise|
+        exercise.update(median_wait_time: median_wait_time(exercise))
+      end
+    end
+
+    private
+    def median_wait_time(exercise)
+      wait_times = Mentor::Discussion.
+        joins(:request).
+        select('TIMESTAMPDIFF(SECOND, request.created_at, mentor_discussions.created_at) AS wait_time').
+        where('request.exercise_id': exercise.id).
+        map(&:wait_time)
```

Comment:
This executes the following query:

```sql
SELECT TIMESTAMPDIFF(SECOND, request.created_at, mentor_discussions.created_at) AS wait_time FROM `mentor_discussions` INNER JOIN `mentor_requests` `request` ON `request`.`id` = `mentor_discussions`.`request_id` WHERE `request`.`exercise_id` = 300
```

This query uses two keys and is thus quite fast.

Reply:
The v2 code had a separate class for the shared bit on joining requests with discussions, but the queries are now subtly different so I've opted not to do that. I also don't think it is a real problem due to the good test coverage I've added in this PR and I'd rather optimize this query than deduplicate the code.